dcos/http.py

```diff
@@ -13,7 +13,18 @@
 
 logger = util.get_logger(__name__)
 
-DEFAULT_TIMEOUT = 5
+DEFAULT_READ_TIMEOUT = 180
```

Comment:
It is up to you whether to keep this, but this PR now does two things:
This PR now does at least the following:

1. It uses the configuration variable in more situations.
2. It changes some requests' defaults to 180 from 5 seconds.

If instead, this were limited to just the former, as per https://jira.mesosphere.com/browse/DCOS_OSS-1956, this could be a much simpler change.

I would personally prefer that. Then, changing the default timeout can be done in another PR which can maybe involve @jgehrcke as a reviewer, and a more thorough look into the optimal timeout. That could even look into removing potentially duplicated logic such as https://github.com/dcos/dcos-cli/blob/3402f831528bca6985eaad1f8308c6a911e6036f/cli/dcoscli/node/main.py#L290.

Reply:
What I mean is, consider setting `DEFAULT_READ_TIMEOUT` to `5` here. I will still have to use the configuration option, but that will unblock me.

Reply:
Indeed, the place you've linked could be simplified too. I also would have prefer to have the default read from the config only in `http.request` instead of as deep as `http._request`, but as the `auth` module uses that private function directly I put it there.

Otherwise the changes in this PR are in my view the bare minimum we must do in order to fix the current situation.